feature: WebClient 연결 타임아웃 처리 및 연결 오류 핸들링

hek316 opened the pull request:
- 외부 API 호출 시 연결 타임아웃과 소켓 타임아웃 시간을 단축하여 과도한 스레드 생성 방지
- 연결 실패 시 사용자에게 "연결 중 에러가 발생하였습니다." 메시지를 반환하여 오류 원인 제공
Close #35

hek316 committed:
feature: WebClient 연결 타임아웃 처리 및 연결 오류 핸들링

hek316 committed:
refactor: GlobalExceptionHandler - 접근제어자 범위 최소화

f-lab-cacio reviewed (approved):
connect_timeout / so_timeout 이 각각 어떤 과정에서의 timeout 을 검증하는 걸까요??

hek316 commented:
> so_timeout

tcp 은 3-way-handshake 라는 과정을 통해서 연결 수립 후 데이터를 주고 받습니다.

`connect_timeout` 은 tcp 의 3-way-handshake 연결 수립 중 패킷 유실이 있다면 패킷을 재전송해야하기  때문에 연결지연이 될 수 있습니다. 여기서 패킷 유실로 인한 어느 정도의 연결지연까지 허용할것인지가 connection timeout 입니다.


`so_timeout` 은 소켓에서 데이터를  수신 시 대기 할 최대 시간을 의미합니다.
클라이언트와 서버의 연결 수립 후 발생합니다. 서버가 응답을 여러개의 패킷으로 나누어 보낼 경우 패킷 간의 지연 즉 각 패킷 사이의 시간차이를 소켓 타임아웃이라고 합니다.

hek316 committed:
refactor: WebClient 타임아웃 설정 개선

- ReadTimeoutHandler와 WriteTimeoutHandler를 추가하여 읽기/쓰기 타임아웃 세밀하게 제어
- 응답 타임아웃(responseTimeout)을 10초로 추가하여 응답 처리 강화
